tests/test_collection.py

```diff
@@ -293,8 +293,7 @@ def test_find_int_float(self):
         self.assertEqual(self.c.find_one({'a': {'$type': 'float'}})['_id'], id_float)
 
     def test_find_with_dots(self):
-        with warnings.catch_warnings(record=True) as w:
-            warnings.simplefilter('once')
+        with self.assertRaises(InvalidKeyError):
```

Comment:
Should this `with` surround each of the individual line(s) that are expected to fail? I think this test passes if "at least one" error is raised, but this should raise multiple errors if I understand it correctly.

Reply:
I tried to touch the unit test as little as possible, simply switching the warnings catch with the assertion. You are right that only one particular line is expected to trigger the exception. We can make that more explicit.